allwinner/address_test.go

```diff
@@ -0,0 +1,85 @@
+package allwinner
+
+import (
+	"os"
+	"path"
+	"testing"
+)
+
+func createDirs(root string, dirs ...string) string {
+	for _, dir := range dirs {
+		if err := os.MkdirAll(path.Join(root, dir), os.ModePerm); err != nil {
+			panic(err)
+		}
+	}
+	return root
+}
+
+func createFiles(root string, paths ...string) string {
+	for _, path_ := range paths {
+		if file, err := os.Create(path.Join(root, path_)); err != nil {
+			panic(err)
```

Comment:
An usual way to do this is to pass t to the function and call t.Fatal(err) here. It's easier to manage.

Reply:
fixed.  thanks, it's been a (very) long time since I have programmed in Go.